pkg/readers/container_image_reader.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/google/osv-scalibr/converter"
+	"github.com/google/osv-scalibr/extractor"
+
+	//"github.com/google/osv-scalibr/converter"
+	el "github.com/google/osv-scalibr/extractor/filesystem/list"
+	sl "github.com/google/osv-scalibr/extractor/standalone/list"
+	"github.com/safedep/vet/pkg/common/logger"
+	"github.com/safedep/vet/pkg/models"
+	"github.com/safedep/vet/pkg/parser"
+)
+
+type ContainerImageReaderConfig struct {
+	Image string
+}
+
+type containerImageReader struct {
+	config *ContainerImageReaderConfig
+}
+
+var _ PackageManifestReader = &containerImageReader{}
+
+func NewContainerImageReader(config *ContainerImageReaderConfig) (*containerImageReader, error) {
+	return &containerImageReader{
+		config: config,
+	}, nil
+}
+
+func (c containerImageReader) Name() string {
+	return "Container Image Reader"
+}
+
+func (c containerImageReader) ApplicationName() (string, error) {
+	return defaultApplicationName, nil
+}
+
+func (c containerImageReader) EnumManifests(handler func(*models.PackageManifest, PackageReader) error) error {
+	image, err := c.getScalibrContainerImage()
+	if err != nil {
+		logger.Errorf("failed to get Scalibr container image: %s", err)
+		return fmt.Errorf("failed to get Scalibr container image: %s", err)
+	}
+
+	scanConfig, err := c.getScalibrScanConfig()
+	if err != nil {
+		logger.Errorf("failed to get scan config: %s", err)
+		return fmt.Errorf("failed to get Scalibr scan config: %s", err)
+	}
+
+	// Scan Container
+	result, err := scalibr.New().ScanContainer(context.Background(), image, scanConfig)
+	if err != nil {
+		logger.Errorf("failed to perform osv-scalibr scan: %s", err)
+		return fmt.Errorf("failed to perform osv-scalibr scan: %s", err)
+	}
+
+	// Handle duplicated packages as same packages (name+version) are found on different locations.
+	// For golang:alpine, 25 % of packages were repeated: the total was 474 with a unique 355
+	packageSet := make(map[string]*extractor.Package)
+
+	manifests := make(map[string]*models.PackageManifest)
+
+	for _, pkg := range result.Inventory.Packages {
+		pkgPurl := converter.ToPURL(pkg)
+		packageSet[pkgPurl.String()] = pkg
+	}
+
+	for _, pkg := range packageSet {
+		if _, ok := manifests[pkg.Ecosystem()]; !ok {
+			manifests[pkg.Ecosystem()] = models.NewPackageManifestFromLocal("", pkg.Ecosystem())
+		}
+
+		pkgDetail := models.NewPackageDetail(pkg.Ecosystem(), pkg.Name, pkg.Version)
+		pkgPackage := &models.Package{
+			PackageDetails: pkgDetail,
+			Manifest:       manifests[pkg.Ecosystem()],
+		}
+
+		manifests[pkg.Ecosystem()].AddPackage(pkgPackage)
+	}
+
+	// TODO: Some Ecosystem is very bad, like for alpine packages the ecosystem is Alpine2.25,
+	for _, manifest := range manifests {
+		err = handler(manifest, NewManifestModelReader(manifest))
+		if err != nil {
+			logger.Errorf("failed to process manifest: %s", err)
+			return fmt.Errorf("failed to process manifest: %s", err)
```

Comment:
```suggestion
			continue
```

Don't fail. Only log error.